### PR TITLE
[AND-169] Apkfy client badge

### PR DESCRIPTION
**Pull request by G0retZ**
**What does this PR do?**

   Adds trusted/certified badge to the Apkfy bottom sheet for the corresponding apps. And only if the corresponding feature flag is enabled 

   Currently missing the actual feature flags repository implementation.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] By commits

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [AND-160](https://aptoide.atlassian.net/browse/AND-160)

**What are the relevant tickets?**

  Tickets related to this pull-request: [AND-169](https://aptoide.atlassian.net/browse/AND-169)

**Related PRs:** [AND-163](#2117)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[AND-160]: https://aptoide.atlassian.net/browse/AND-160?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AND-169]: https://aptoide.atlassian.net/browse/AND-169?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AND-163]: https://aptoide.atlassian.net/browse/AND-163?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app-games/src/main/java/com/aptoide/android/aptoidegames/apkfy/ApkfyBottomSheetContent.kt [hunk 1]
```diff
@@ -43,7 +61,62 @@ class ApkfyBottomSheetContent(private val app: App) : BottomSheetContent {
...
+fun Badge(app: App) {
+  val state = rememberBadgeVisibility()
+  when {
+    state && (app.isInCatappult() == true) -> Row(
+      verticalAlignment = Alignment.CenterVertically,
+      modifier = Modifier
+        .background(color = Palette.Official)
+        .padding(horizontal = 8.dp, vertical = 4.dp)
+    ) {
+      getVerifiedIcon(Palette.White)
```
Comment (pedroagostinhoaptoide) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/apkfy/ApkfyBottomSheetContent.kt:81]: The Image or Icon composable is missing around the vector

**Review by pedroagostinhoaptoide — CHANGES_REQUESTED** [review]
Should the PR base change to #2118 ?

File: app-games/src/main/java/com/aptoide/android/aptoidegames/apkfy/presentation/BadgeStateProvider.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+fun rememberBadgeVisibility(): Boolean = runPreviewable(
+  preview = { Random.nextBoolean() },
+  real = {
+    val coroutineScope = rememberCoroutineScope()
+    var state by remember { mutableStateOf(false) }
+    val vm = hiltViewModel<InjectionsProvider>()
+    LaunchedEffect(key1 = Unit) {
+      coroutineScope.launch {
+        state = vm.featureFlags.getFlag("apkfy_badge", false)
+        println("Boom $state")
```
Comment (pedroagostinhoaptoide) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/apkfy/presentation/BadgeStateProvider.kt:34]: Remove unused print

Comment (G0retZ) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/apkfy/presentation/BadgeStateProvider.kt:34]: Fixed

**Review by pedroagostinhoaptoide — APPROVED** [review]

**Pull request merged**